integration/src/test/scala/com/snowplowanalytics/snowplow/collectors/scalastream/integration/utils/Containers.scala

```diff
@@ -0,0 +1,104 @@
+ * IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.collectors.scalastream.integration.utils
+
+import cats.effect.{Resource, Sync}
+import com.dimafeng.testcontainers.GenericContainer
+import com.snowplowanalytics.snowplow.collectors.scalastream.integration.CollectorConfig
+import org.slf4j.LoggerFactory
+import org.testcontainers.containers.output.Slf4jLogConsumer
+import org.testcontainers.containers.wait.strategy.Wait
+import org.testcontainers.containers.{BindMode, Network, GenericContainer => JGenericContainer}
+import org.testcontainers.images.builder.ImageFromDockerfile
+
+object Containers {
+  val LocalstackExposedPort = 4566
+  val CollectorExposedPort  = 12345
+
+  private val network = Network.newNetwork()
+
+  def localstack: JGenericContainer[_] = {
+    val container = GenericContainer(
+      dockerImage  = "localstack/localstack-light:1.2.0",
+      exposedPorts = Seq(LocalstackExposedPort),
+      env          = Map("USE_SSL" -> "1"),
+      waitStrategy = Wait.forLogMessage(".*AWS kinesis.CreateStream.*", 2),
+      fileSystemBind = Seq(
+        GenericContainer.FileSystemBind(
+          "./.localstack",
+          "/var/lib/localstack",
+          BindMode.READ_WRITE
+        ),
+        GenericContainer.FileSystemBind(
+          "integration/src/test/resources/localstack",
+          "/docker-entrypoint-initaws.d",
+          BindMode.READ_ONLY
+        )
+      )
+    )
+    container.underlyingUnsafeContainer.withNetwork(network)
+    container.underlyingUnsafeContainer.withNetworkAliases("localstack")
+    container.container
+  }
+
+  def collector(
+    flavour: String,
+    testConfig: CollectorConfig,
+    dependsOn: Option[JGenericContainer[_]] = None
+  ): JGenericContainer[_] = {
+    val imageFromDockerfile = new ImageFromDockerfile()
+      .withDockerfile(java.nio.file.Path.of(flavour, "target", "docker", "stage", "Dockerfile"))
+    val container = GenericContainer(
+      dockerImage  = imageFromDockerfile,
+      exposedPorts = Seq(CollectorExposedPort),
+      env          = Map("AWS_ACCESS_KEY_ID" -> "test", "AWS_SECRET_KEY" -> "test") ++ testConfig,
+      command      = Seq("--config", s"/snowplow/config/collector.hocon"),
+      waitStrategy = Wait.forLogMessage(".*REST interface bound to.*", 1),
+      fileSystemBind = Seq(
+        GenericContainer.FileSystemBind(
+          s"integration/src/test/resources/collector_config/collector.hocon",
+          "/snowplow/config/collector.hocon",
+          BindMode.READ_ONLY
+        )
+      )
+    )
+    container.underlyingUnsafeContainer.withNetwork(network)
+    container.underlyingUnsafeContainer.withNetworkAliases("collector")
+    dependsOn.foreach(container.underlyingUnsafeContainer.dependsOn(_))
```

Comment:
If we make the collector depend on `localstack` container then I think that we don't need to start `localstack` manually, because starting the collector would do it. I let you check that. Then if we keep this `dependsOn` and stop starting `localstack` ourselves (in the `synchronized`), it's important to figure out if we're sure that several instances of `localstack` are never started in parallel.